tidb-server/main.go

```diff
@@ -134,59 +134,115 @@ const (
 )
 
 var (
-	version      = flagBoolean(nmVersion, false, "print version information and exit")
-	configPath   = flag.String(nmConfig, "", "config file path")
-	configCheck  = flagBoolean(nmConfigCheck, false, "check config file validity and exit")
+	version      *bool
+	configPath   *string
+	configCheck  *bool
+	configStrict *bool
+
+	// Base
+	store            *string
+	storePath        *string
+	host             *string
+	advertiseAddress *string
+	port             *string
+	cors             *string
+	socket           *string
+	enableBinlog     *bool
+	runDDL           *bool
+	ddlLease         *string
+	tokenLimit       *int
+	pluginDir        *string
+	pluginLoad       *string
+	affinityCPU      *string
+	repairMode       *bool
+	repairList       *string
+	tempDir          *string
+
+	// Log
+	logLevel     *string
+	logFile      *string
+	logSlowQuery *string
+
+	// Status
+	reportStatus    *bool
+	statusHost      *string
+	statusPort      *string
+	metricsAddr     *string
+	metricsInterval *uint
+
+	// PROXY Protocol
+	proxyProtocolNetworks      *string
+	proxyProtocolHeaderTimeout *uint
+	proxyProtocolFallbackable  *bool
+
+	// Bootstrap and security
+	initializeSecure            *bool
+	initializeInsecure          *bool
+	initializeSQLFile           *string
+	disconnectOnExpiredPassword *bool
+	keyspaceName                *string
+	serviceScope                *string
+	help                        *bool
+)
+
+func initflag() {
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
```

Comment:
What about just `fpset := flag.NewFlagSet`?

Reply:
In fact, the global one have been polluted by the glog. so we need to clean it.

Reply:
I try it. it is unavailable.

Reply:
I try it again. it works.